src/main/java/com/minecolonies/coremod/entity/ai/mobs/AbstractArcherBarbarian.java

```diff
@@ -0,0 +1,302 @@
+import net.minecraft.world.DifficultyInstance;
+import net.minecraft.world.EnumDifficulty;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import javax.annotation.Nullable;
+import java.util.Calendar;
+
+/**
+ * Abstract Class implementing the Archer Barbarian
+ */
+public abstract class AbstractArcherBarbarian extends EntityMob implements IRangedAttackMob
+{
+    /* default */ private final        Colony                           colony                = ColonyManager.getClosestColony(world, this.getPosition());
+    /* default */ private static final DataParameter<Boolean>           SWINGING_ARMS         =
+      EntityDataManager.<Boolean>createKey(AbstractArcherBarbarian.class, DataSerializers.BOOLEAN);
+    /* default */ private final        EntityAIAttackRangedBowBarbarian aiArrowAttack         = new EntityAIAttackRangedBowBarbarian(this, 1.0D, 20, 15.0F);
+    /* default */ private static final float                            ENTITY_WIDTH          = 0.6F;
+    /* default */ private static final float                            ENTITY_HEIGHT         = 1.99F;
+    /* default */ private static final int                              BARBARIAN_BASE_HEALTH = 25;
+    /* default */ private static final int                              DIFFICULTY_SUBTRACTER = 14;
+    /* default */ private static final int                              DIFFICULTY_MULTIPLIER = 4;
+    /* default */ private static final int                              BOW_DAMAGE            = 20;
+    /* default */ private static final int                              BOW_DAMAGE_HARD       = 40;
+    /* default */ private static final double                           GRAVITY_MULTIPLIER    = 0.20000000298023224D;
+    /* default */ private static final float                            ARROW_VELOCITY        = 1.6F;
+    /* default */ private static final float                            SOUND_MULTIPLIER      = 0.4F;
+    /* default */ private static final float                            SOUND_ADDER           = 0.8F;
+    /* default */ private static final float                            ENTITY_EYE_HEIGHT     = 1.74F;
+
+    private final EntityAIAttackMelee aiAttackOnCollide = new EntityAIAttackMelee(this, 1.2D, false)
+    {
+        /**
+         * Resets the task
+         */
+        public void resetTask()
+        {
+            super.resetTask();
+            AbstractArcherBarbarian.this.setSwingingArms(false);
+        }
+
+        /**
+         * Execute a one shot task or start executing a continuous task
+         */
+        public void startExecuting()
+        {
+            super.startExecuting();
+            AbstractArcherBarbarian.this.setSwingingArms(true);
+        }
+    };
+
+    public AbstractArcherBarbarian(final World worldIn)
+    {
+        super(worldIn);
+        this.setSize(ENTITY_WIDTH, ENTITY_HEIGHT);
+        this.setCombatTask();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this call from a constructor to the overridable "setCombatTask" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)